guessing_game.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry-byebug'
+
+class GuessingGame
+
+  attr_accessor :congrats_message, :remaining_guesses, :won
+
+  def initialize(secret_number, num_guesses)
+    @congrats_message = "Yay, you won!"
+    @secret_number = secret_number
+    @remaining_guesses = num_guesses
+    @guessed_numbers = []
+    @won = false
+    @lost = false
+  end
+
+  def has_won?
+    @won
+  end
+
+  def has_lost?
+    @lost
+  end
+
+  def guess(num)
+    return "You already won. The number was #{@secret_number}" if has_won?
+    return "You already lost. The number was #{@secret_number}" if has_lost?
+
+    if num < @secret_number
+      deduct_guesses(num)
+      warning("Too low!")
+    elsif num > @secret_number
+      deduct_guesses(num)
+      warning("Too high!")
+    else num == @secret_number
+      deduct_guesses(num)
```

Comment:
if you `deduct_guesses(num)` in all 3 cases, you can move that statement before all the `if` statements